src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -184,6 +184,8 @@ protected AbstractFolder(ItemGroup parent, String name) {
     protected void init() {
         if (properties == null) {
             properties = new DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor>(this);
+        } else {
+            properties.setOwner(this);
```

Comment:
During `readResolve`, `this` will in general be incomplete in various ways. Unsafe to pass an instance to plugin code which might expect to do things like call `owner.getParent()`, etc. `onLoad` is called later when everything is wired up.